adapters/repos/db/vector/hnsw/maintainance.go

```diff
@@ -66,7 +66,7 @@ func growIndexToAccomodateNode(index []*vertex, id uint64,
 	// typically grow the index by the delta
 	newSize := previousSize + defaultIndexGrowthDelta
 
-	if uint64(newSize) < id {
+	if uint64(newSize) <= id {
```

Comment:
wow pretty amazing that you were able to spot this change on first sight yesterday 🎉